src/libraries/Common/src/System/Net/Sockets/SocketConnection.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Connections
+{
+    internal sealed class SocketConnection : Connection, IConnectionProperties
+    {
+        private readonly Socket _socket;
+        private readonly SocketsConnectionFactory? _factory;
+        private readonly IConnectionProperties? _options;
+        private Stream? _stream;
+#if SYSTEM_NET_SOCKETS_DLL
```

Comment:
Can we add a comment explaining this? (It's not clear to me what we're doing here)

Reply:
Will add comment, the internal class is being reused as part of System.Net.Http's synchronous logic originally added in #39524.

Reply:
If it's just for the sync logic, can we just have a separate implementation of Connection that's like InternalSyncSocketConnection that lives in System.Net.Http? Seems like it's ~10 lines of code or so, right?